commands/marketplace_publish.go

```diff
@@ -110,9 +105,9 @@ func (c *marketplacePublishCmd) runE(cmd *cobra.Command, args []string) error {
 		return err
 	}
 	fmt.Printf("%s Service published with success\n", pretty.SuccessSign)
-	fmt.Printf("%s See it on the marketplace: https://marketplace.mesg.com/services/%s\n", pretty.SuccessSign, c.manifest.Service.Definition.Sid)
+	fmt.Printf("%s See it on the marketplace: https://marketplace.mesg.com/services/%s\n", pretty.SuccessSign, c.sid)
 
-	fmt.Printf("%s To create a service offer, execute the command:\n\tmesg-core marketplace create-offer %s\n", pretty.SuccessSign, c.manifest.Service.Definition.Sid)
+	fmt.Printf("%s To create a service offer, execute the command:\n\tmesg-core marketplace create-offer %s\n", pretty.SuccessSign, c.sid)
```

Comment:
`c.sid` is always an empty string. You forgot to set its value after the deploy.

Reply:
Same for `c.hash`

From:
```
	sid, hash, err := deployService(c.e, c.path, map[string]string{})
```
to
```
	c.sid, c.hash, err := deployService(c.e, c.path, map[string]string{})
```

https://github.com/mesg-foundation/core/pull/837/files#diff-18a1eba1f644f5f71dc449b67f9e1f24R50

Reply:
good catch, fixed